### ISSUE TITLE
Non-framework cross validation

### ISSUE THREAD

**Issue by lisitsyn** [shogun-toolbox/shogun#2740]
If you compare how you do xval in sklearn and shogun there is a significant difference - sklearn's xval just gives you indices for train/test set while we do it as framework. Probably it would be better if we follow the same principle as it gives much more flexibility (what to evaluate, what to train and how) even though it takes more code to write.

@karlnapf @sonney2k @lambday @vigsterkr discuss?

**Comment by karlnapf** [comment]
Our framework also can give indices.

I think we should include the cross-validation of methods within Shogun, it is very useful and easy to get wrong for people who are not familiar.

However, brute force parameter (grid) search should be out, as this can easily be done with a nested loop given that x-validation is implemented.

**Comment by lisitsyn** [comment]
Closing for now.
